transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -261,6 +261,29 @@ def _create_transpose(self):
         self._transpose = tex.fp8_transpose(data, self._fp8_dtype, out=self._transpose)
         self._transpose_invalid = False
 
+    def _fix_gathered_transpose(self, tp_size=1, from_rowwise=False):
```

Comment:
I don't like how unintuitive and UB-specific this function is. It would have been better to implement `CommOverlap::get_buffer` so that it internally transposes the UB buffer and constructs a `Float8Tensor` with only column-wise data (instead of how this moves the row-wise data to the column-wise data). We're not quite there yet since `Float8Tensor`s with only column-wise data experience problems.

For now, we should delete this function and reimplement the logic in the linear modules. We should make clear that this is a hacky workaround and not allow it to become an API we rely on.